### ISSUE TITLE
 Space in search BaseDN causes request to fail

### ISSUE THREAD

**Issue by ghost** [glauth/glauth#68]
I wanted to try out GLAuth instead of OpenLDAP for use with Authelia. GLAuth works perfectly with ldaptools, but with Authelia I always get "Authentication failed", so I started to debug the requests and found out Authelia puts a space between the request BaseDN so instead of `dc=example,dc=com` it sends `dc=example, dc=com` which causes the request to fail. I wasn't sure whether to submit this to GLAuth or to Authelia, so I'll just submit it to both.

https://github.com/clems4ever/authelia/issues/306

Example of working request (done with ldapsearch):

```
14:13:12.879190 Bind ▶ DEBU 054 Bind request: bindDN: cn=admin,ou=admins,dc=example,dc=com, BaseDN: dc=example,dc=com, source: 192.168.0.68:54638
14:13:12.879279 Bind ▶ DEBU 055 Bind success as %s from %s cn=admin,ou=admins,dc=example,dc=com 192.168.0.68:54638
14:13:12.879753 Search ▶ DEBU 056 Search request as %s from %s for %s cn=admin,ou=admins,dc=example,dc=com 192.168.0.68:54638 (objectclass=*)
14:13:12.879955 Search ▶ DEBU 057 AP: Search OK: %s (objectclass=*)
```

Example of broken request

```

14:13:27.437198 Bind ▶ DEBU 058 Bind request: bindDN: cn=admin,ou=admins,dc=example,dc=com, BaseDN: dc=example,dc=com, source: 192.168.0.68:54642
14:13:27.437283 Bind ▶ DEBU 059 Bind success as %s from %s cn=admin,ou=admins,dc=example,dc=com 192.168.0.68:54642
14:13:27.437730 Search ▶ DEBU 05a Search request as %s from %s for %s cn=admin,ou=admins,dc=example,dc=com 192.168.0.68:54642 (objectclass=*)
2018/12/15 14:13:27 handleSearchRequest error LDAP Result Code 50 "Insufficient Access Rights": Search Error: search BaseDN dc=example, dc=com is not in our BaseDN dc=example,dc=com
```

**Comment by mgrandl** [comment]
Well I haven't either, it's just that Authelia appears to put one for the request in there. OpenLDAP seems to not have a problem with it. The configuration is basically just the sample-simple.cfg with only one user and group to try it out.

**Comment by ryskov** [comment]
Would it break LDAP specifications if GLAuth were to remove spaces on the incoming search basedn before processing? Would be pretty simple to implement and if it could help people move things over to GLAuth i think its a valid enough argument to do it.

**Comment by SISheogorath** [comment]
Same problem over here. Seems like the RFCs solved htat question: https://tools.ietf.org/html/rfc4517#appendix-B see 25

**Comment by darkguy2008** [comment]
Well, mine doesn't have any spaces either and I'm getting the same error, as if for a search the BaseDN is always empty, makes no sense.
```
glauth_1  | 2019/05/12 12:24:27 handleSearchRequest error LDAP Result Code 50 "Insufficient Access Rights": Search Error: search BaseDN  is not in our BaseDN dc=domain,dc=com
```

**Comment by edslatt** [comment]
I'm also showing an empty BaseDN using a client that produces proper results against other servers.

I really want to use glauth because its so light but I need it to work this this client - GitLab

**Comment by Fusion** [comment]
I believe this was fixed by https://github.com/glauth/glauth/pull/133 anyone care to confirm?

**Comment by Fusion** [comment]
According to my testing, fixed by #133 (do not trust that PR's name! It's pretty big due to code drift...)

**Comment by sperolsmicroej** [comment]
This problem still exists
